lib/Dialect/Visitor.cpp

```diff
@@ -51,37 +51,58 @@ void VisitorTemplate::add(VisitorKey key, VisitorCallback *fn,
   handler.callback = fn;
   handler.data = data;
   handler.projection = projection;
+
   m_handlers.emplace_back(handler);
 
-  unsigned handlerIdx = m_handlers.size() - 1;
+  const unsigned handlerIdx = m_handlers.size() - 1;
 
   if (key.m_kind == VisitorKey::Kind::Intrinsic) {
-    m_intrinsicIdMap[key.m_intrinsicId].push_back(handlerIdx);
-  } else {
+    m_opMap[OpDescription::fromIntrinsic(key.m_intrinsicId)].push_back(
+        handlerIdx);
+  } else if (key.m_kind == VisitorKey::Kind::OpDescription) {
     const OpDescription *description = key.m_description;
     switch (description->getKind()) {
     case OpDescription::Kind::Core:
-      for (unsigned opcode : description->getOpcodes())
-        m_coreOpcodeMap[opcode].push_back(handlerIdx);
+      for (unsigned op : description->getOpcodes())
+        m_opMap[OpDescription::fromCoreOp(op)].push_back(handlerIdx);
```

Comment:
It should be possible to simplify this entire block of code down to `m_opMap[*description].push_back(handleIdx);`